lib/pipeline.js

```diff
@@ -499,6 +487,50 @@ class PipelineModel extends BaseModel {
         return admin;
     }
 
+    async getFirstAdmin() {
+        /* eslint-disable no-restricted-syntax */
+        for (const username of Object.keys(this.admins)) {
+            try {
+                /* eslint-disable no-await-in-loop */
+                return await new Promise((resolve, reject) => {
+                /* eslint-disable no-await-in-loop */
+
+                    // Lazy load factory dependency to prevent circular dependency issues
+                    // https://nodejs.org/api/modules.html#modules_cycles
+                    /* eslint-disable global-require */
+                    const UserFactory = require('./userFactory');
+                    /* eslint-enable global-require */
+                    const factory = UserFactory.getInstance();
+
+                    const user = factory.get({
+                        username,
+                        scmContext: this.scmContext
+                    });
+
+                    user.then((realUser) => {
+                        realUser.getPermissions(this.scmUri)
+                            .then((permissions) => {
+                                if (!permissions.push) {
+                                    delete this.admins[username];
+
+                                    return reject();
+                                }
+
+                                return resolve(realUser);
+                            });
+                    });
+                });
+            } catch (e) {
+                /* eslint-disable no-console */
+                console.log(e);
+                /* eslint-disable no-console */
+            }
+        }
+        /* eslint-disable no-restricted-syntax */
+
+        return false;
```

Comment:
Is this line needed or Is `return false` enough?

Reply:
I think `null` would make more sense in this scenario.

Reply:
I add `throw new Error('Pipeline has no admin')`. I think this make it easy to understand what happened.